internal/backend/backend.go

```diff
@@ -122,6 +124,31 @@ func (b *Backend) RemoveUser(ctx context.Context, userID string, removeFiles boo
 	return nil
 }
 
+func (b *Backend) GetCounts(ctx context.Context, userID string) (map[imap.MailboxID]int, error) {
+	b.usersLock.Lock()
+	defer b.usersLock.Unlock()
+
+	return db.ReadResult(ctx, b.users[userID].db, func(ctx context.Context, c *ent.Client) (map[imap.MailboxID]int, error) {
+		counts := make(map[imap.MailboxID]int)
+
+		mailboxes, err := c.Mailbox.Query().All(ctx)
```

Comment:
Add select field with ID only to avoid loading more data than necessary.

Reply:
done